rclc/CMakeLists.txt

```diff
@@ -4,8 +4,28 @@ project(rclc)
 
 find_package(ament_cmake REQUIRED)
 find_package(rcl REQUIRED)
+find_package(rosidl_generator_c REQUIRED)
 
 ament_export_include_directories(include)
+include_directories(include)
+
+set(rclc_sources
+  "src/functions.c"
+)
+set_source_files_properties(${rclc_sources} PROPERTIES LANGUAGE "C")
+add_library(${PROJECT_NAME} ${rclc_sources})
```

Comment:
ROS 2 expects SHARED libraries by default.
To respect the CMake option "BUILD_SHARED_LIBS" we encourage packages to depend on `ament_cmake_ros` to be compiled statically or dynamically according to the CMake option. See https://github.com/ros2/rcl/pull/186/files for an example of converting a package into an `ament_cmake_ros` package